client/related_items/ComparisonModal.jsx

```diff
@@ -11,13 +11,37 @@ class ComparisonModal extends React.Component {
     }
 
     let key = 0;
+    let mainFeatures = this.props.mainProduct[1];
+    let comparisonFeatures = this.props.productToCompare[1];
+    let allFeatures = [];
+
+    mainFeatures.forEach(feature => {
+      feature.compValue = '';
+      allFeatures.push(feature);
+    });
+
+    comparisonFeatures.forEach(compFeature => {
+      var index = allFeatures.findIndex(item => item.feature === compFeature.feature);
```

Comment:
Had to look this .findIndex up. Nice one!

Reply:
Haha this was after hours of banging my head against the wall trying to figure this out, and forgot Arrays have all sorts of helpful methods.